ERCS/erc-7715.md

```diff
@@ -0,0 +1,551 @@
+
+```
+struct Action {
+    address to;
+    uint256 value;
+    bytes data;
+}
+```
+
+A simple pseudocode example of using a permission in this way, given two ethers signers in the same context, where `alice` wants to request a permission from `bob` might be like this:
+
+```
+// Alice requests a permission from Bob
+const permissionsResponse = await bob.request({
+  method: 'wallet_grantPermissions',
+  params: [{
+    address: bob.address,
+    chainId: 123,
+    signer: {
+      type: 'account',
+      data: {
+        id: alice.address
+      }
+    },
+    permissions: [{
+      type: 'native-token-transfer',
+      policies: [{
+        type: 'gas-limit',
+        data: '0x0186A0'
+      }]
+      data: {
+        allowance: '0x0DE0B6B3A7640000'
+      },
+      required: true
+    }],
+    expiry: Math.floor(Date.now() / 1000) + 3600 // 1 hour from now
+  }]
+});
+
+// Extract the permissionsContext and delegationManager
+const permissionsContext = permissionsResponse.permissionsContext;
+const delegationManager = permissionsResponse.signerMeta.delegationManager;
+
+// Alice forms the action she wants Bob's account to take
+const action = {
+  to: alice.address,
+  value: '0x06F05B59D3B20000'
+  data: '0x'
+};
+
+// Alice sends the transaction by calling redeemDelegation on Bob's account
+const tx = await bob.sendTransaction({
+  to: delegationManager,
+  data: bob.interface.encodeFunctionData('redeemDelegation', [
+    permissionsContext,
+    action
+  ])
+});
+
+```
+
+## Rationale
+
+The typical transaction flow of `suggesting transactions => approving transactions => sending transactions` is deeply limiting in several ways:
+
+- Users must be online to send transactions. DApps cannot send transactions for users when they are offline, which makes use cases such as subscriptions or automated trading impossible.
+
+- Users must manually approve every transaction, interrupting what could otherwise be a smooth user experience.
+
+With this ERC, DApps can request Wallets to grant permissions and execute transactions on the user's behalf, therefore circumventing the issues above.
+
+### `permissionsContext`
+
+Since this ERC only specifies the interaction between the wallet and the DApp but not how the wallet enforces permissions, we need a flexible way for the wallet to pass along information to the DApp so that it can construct transactions that imbue the permissions.
+
+The `permissionsContext` field is meant to be an opaque string that's maximally flexible and can encode arbitrary information for different permissions schemes. We specifically had three schemes in mind:
+
+- If a DApp leverages [ERC-7679](./eip-7679.md), it could use `permissionsContext` as the `context` parameter when interacting with the UserOp builder.
```

Comment:
```suggestion
- If a DApp leverages [ERC-7679](./erc-7679.md), it could use `permissionsContext` as the `context` parameter when interacting with the UserOp builder.
```